DemoApp/SchibstedAccountDemo/AppDelegate.swift

```diff
@@ -132,18 +134,56 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         }
     }
 
-    func application(_: UIApplication, didFinishLaunchingWithOptions _: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
-        self.user = IdentityManager(clientConfiguration: .current).currentUser
+    func application(_: UIApplication, didFinishLaunchingWithOptions options: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
+        self.user = self.identityManager.currentUser
 
         AppDelegate.pulseTracker.track([
             "demo-app-launch": [
                 "user": String(describing: self.user),
             ],
         ])
 
+        let doesLaunchOptionsContainRecognizedURL = AppLaunchData(launchOptions: options, clientConfiguration: .current) != nil
+        if !doesLaunchOptionsContainRecognizedURL, self.user?.state == .loggedIn {
+            self.ensureAcceptanceOfNewTerms()
+            return true
+        }
+
         return true
     }
 
+    private func ensureAcceptanceOfNewTerms() {
+        self.user?.agreements.status { [weak self] result in
+            switch result {
+            case let .success(hasAcceptedLatestTerms):
+                if hasAcceptedLatestTerms {
+                    // Latest terms already accepted, nothing else to do.
+                    return
+                }
+
+                // Fetch the latest terms.
+                self?.identityManager.fetchTerms { [weak self] result in
+                    switch result {
+                    case let .success(terms):
+                        // Present UI to accept new terms.
+                        guard let viewController = self?.window?.rootViewController, let user = self?.identityManager.currentUser else {
+                            return
+                        }
+                        // It is important that you pass the same instance of `User` that you previously stored, otherwise you won't get logout notifications
+                        // for that user in case the user is logged out for not having accepted the new terms.
+                        IdentityUI.presentTerms(terms, for: user, from: viewController, configuration: .current)
```

Comment:
User has an internal configuration object as well, so it's probably better to just use that one?

Reply:
Ah, it's a UI config, not a client config :) 

Should this be an error if `user.clientCOnfiguration != configuration.clientConfiguration` => `.failure(something)` ?

Reply:
Oh, good point! Ideally, I would really prefer enforcing a coherent state through the APIs rather than trowing an error if the client makes a mistake, but it doesn't look trivial in this case.

Btw, is there a specific reason why the client configuration is owned by the UI configuration, or could they be treated more independently? Either way that is of course out of scope, so I guess an error has to be ok for now.

Reply:
No particular reason other than it felt more friendly to have client pass one configuration object instead of two.